blog/_posts/GSoC 2017 project A Wrapper for the FEniCS Finite Element Toolbox.md

```diff
@@ -0,0 +1,107 @@
+```julia
+using FEniCS
+mesh = UnitSquareMesh(8,8) 
+V = FunctionSpace(mesh,"P",1)
+u_D = Expression("1+x[0]*x[0]+2*x[1]*x[1]", degree=2)
+u = TrialFunction(V)
+bc1 = DirichletBC(V,u_D, "on_boundary")
+v = TestFunction(V)
+f = Constant(-6.0)
+a = dot(grad(u),grad(v))*dx
+L = f*v*dx
+U = FEniCS.Function(V)
+lvsolve(a,L,U,bc1) #linear variational solver
+errornorm(u_D, U, norm="L2")
+get_array(L) #this returns an array for the stiffness matrix
+get_array(U) #this returns an array for the solution values
+vtkfile = File('poisson/solution.pvd')
+vtkfile << U.pyobject #exports the solution to a vtkfile
+
+```
+Apart from just defining the problem, we can also access and save the arrays corresponding to various variational forms. We can do this as follows :
+
+```julia
+a = dot(grad(u),grad(v))*dx #this sets up the variatonal form from the previous problem
+variational_matrix = get_array(a)
+
+```
+
+We can also plot the solution (this relies on FEniCS backend for plotting) :
+
+```julia
+FEniCS.Plot(U)
+FEniCS.Plot(mesh)
+
+```
+
+![alt text](https://github.com/ysimillides/FEniCS.jl/blob/master/examples/result.png "Solution")
+ 
+![alt text](https://github.com/ysimillides/FEniCS.jl/blob/master/examples/mesh.png "Square Mesh")
+
+
+
+
+# Challenges 
+
+Due to the nature of this project, which relied on FEniCS, we faced various challenges throughout the summer. These included, but where not limited to build errors, where various parts of the package failed to compile, to unexpected errors in the usability of the code. Chris and Bart where very helpful, in both pointing these out, and in assisting in fixing them. In some parts the documentation was slightly patchy which also complicated parts of the project as some functions where ambiguous towards their intended use. 
+
+# Further Improvements
+
+I hope to be able to maintain and improve the package, using it where possibly throughout my further studies. Some identifiable improvements, in order of difficulty are : 
+
+- Fixing precompilation which would provide a large performance benefit. This error is well documented, and the fix is relatively simple. At the same time it would require the rewriting of a large segment of the codebase due to the way we currently access functions and attributes. 
+- Improving plotting. We currently rely on the FEniCS plotting backend to plot the necessary functions/meshes/objects. For more detailed visualization we can use Paraview, like in FEniCS. A direct Julia plotter would be nice, as we could provide further customization to our objects.
+- Integration with JuliaDiffEq. We can currently specify and create the necessary objects for the solution of some FEM problems. We have also provided interfaces for accessing most of their attributes aswell as exporting the necessary arrays. Despite this, we currently have no automatic way of seamlessly accessing them via other packages. By providing this access, we would be able to greatly extend the packages capabilities.
+
+ 
```

Comment:
Maybe add a comment here about the FEniCS form compiler, which takes e.g. a matrix assembly expression and compiles this to C code and then machine code. Clearly, it would be an opportunity to replace this whole chain directly with Julia.